recipes/gtkmm/2.x.x/conanfile.py

```diff
@@ -0,0 +1,80 @@
+import os
+
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+
+class GtkmmConan(ConanFile):
+    name = "gtkmm"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-2.1-or-later"
+    homepage = "https://www.gtk.org"
+    description = "The C++ API for GTK."
+    settings = {"os": "Linux"}
+    options = {"version": [2]}
+    default_options = {"version": 2}
+    topics = ("gui", "widget", "graphical")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+              }
+    default_options = {"shared": False,
+                       "fPIC": True}
+    generators = "pkg_config"
+    _autoconf = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and \
+           "MT" in str(self.settings.compiler.runtime) and self.options.shared:
+            raise ConanInvalidConfiguration("Visual Studio and Runtime MT is not supported for shared library.")
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def requirements(self):
+        self.requires('gtk/system')
+        self.requires('libsigcpp/3.0.0')
+
+    def _configure_build(self):
+        if self._autoconf:
+            return self._autoconf
+
+        self._autoconf = AutoToolsBuildEnvironment(self)
```

Comment:
gtkmm has meson support.
Is that working?

Reply:
I preferred to use Meson too, but the `meson.build` files in the source repository are not in the official release tar balls. Their README talks about separate release tar balls based on `autoconf` and `meson.build`, but on their own website, I only found the ones supporting autoconf. Ergo, I went with AutoTools for the package.